flink-connectors/flink-connector-kafka-0.9/src/main/java/org/apache/flink/streaming/connectors/kafka/internal/KafkaConsumerThread.java

```diff
@@ -240,7 +249,9 @@ public void run() {
 						newPartitions = unassignedPartitionsQueue.getBatchBlocking();
 					}
 					if (newPartitions != null) {
-						reassignPartitions(newPartitions);
+						reassignPartitions(newPartitions, new HashSet<>());
```

Comment:
I just realized this should be actually
`reassignPartitions(newPartitions, partitionsToBeRemoved);`